sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1639,16 +1639,14 @@ cl_int ExecCGCommand::enqueueImp() {
       case kernel_param_kind_t::kind_accessor: {
         Requirement *Req = (Requirement *)(Arg.MPtr);
         AllocaCommandBase *AllocaCmd = getAllocaForReq(Req);
-#if USE_PI_CUDA
-        pi_mem MemArg = (pi_mem)AllocaCmd->getMemAllocation();
-        Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, Arg.MIndex, &MemArg);
-#else
         RT::PiMem MemArg = (RT::PiMem)AllocaCmd->getMemAllocation();
+        if (!RT::useBackend(pi::Backend::SYCL_BE_PI_OPENCL)) {
+          Plugin.call<PiApiKind::piextKernelSetArgMemObj>(Kernel, Arg.MIndex,
+                                                          &MemArg);
+          break;
+        }
         Plugin.call<PiApiKind::piKernelSetArg>(Kernel, Arg.MIndex,
```

Comment:
I think an else would be a bit clearer than having two breaks, took me a couple of seconds to understand the change on the diff.

Reply:
Refactored, thanks